analytics/views/support.py

```diff
@@ -375,7 +375,8 @@ def support(
                 current_plan=current_plan,
             )
             if current_plan is not None:
-                new_plan, last_ledger_entry = make_end_of_cycle_updates_if_needed(
+                billing_session = RealmBillingSession(user=None, realm=realm)
```

Comment:
I think in this code path, in particular, we can pass `user=acting_user` for the `user` parameter. In this particular function, I think the user isn't really going to be used, but other functions called from this endpoint should possibly have the user so that we can make `RealmAuditLog` entries that record which staff member made a change.

@laurynmm @mateuszmandera FYI for possibly post-merge thoughts on this detail.

Reply:
`make_end_of_cycle_updates_if_needed` has
```py
  RealmAuditLog.objects.create(
      realm=realm,
      event_time=event_time,
      event_type=RealmAuditLog.CUSTOMER_SWITCHED_FROM_MONTHLY_TO_ANNUAL_PLAN,
      extra_data={
          "monthly_plan_id": plan.id,
          "annual_plan_id": new_plan.id,
      },
  )
```
Currently, we log `acting_user=null` for the new rows created in these cases. Do we plan to make the change of using `acting_user` instead of null ?

Reply:
Oh, good point, yeah, I think `acting_user=None` is correct for this function in all cases -- it's a system job, and other code paths that are calling it are just making sure the system job's work is up-to-date before presenting the information to a user, for resilience in case the cron job starts failing or something.